yapf/__init__.py

```diff
@@ -143,11 +143,13 @@ def main(argv):
         original_source.append(py3compat.raw_input())
       except EOFError:
         break
+
+    original_source.append('')
     style_config = args.style
     if style_config is None and not args.no_local_style:
       style_config = file_resources.GetDefaultStyleForDir(os.getcwd())
     reformatted_source, changed = yapf_api.FormatCode(
-        py3compat.unicode('\n'.join(original_source) + '\n'),
+        py3compat.unicode('\n'.join(original_source)),
```

Comment:
It should be, because if you add newline after joining you're creating a new string and when joining you don't